explainability-service/src/main/java/org/kie/trustyai/service/data/storage/MinioStorage.java

```diff
@@ -95,17 +96,60 @@ public ByteBuffer getInputData() throws StorageReadException {
 
     @Override
     public ByteBuffer getOutputData() throws StorageReadException {
-        try {
-            isObjectAvailable(this.bucketName, this.outputFilename);
-        } catch (MinioException e) {
-            LOG.error("Input file '" + this.outputFilename + "' at bucket '" + this.bucketName + "' is not available");
-            throw new StorageReadException(e.getMessage());
-        }
+        isObjectAvailable(this.bucketName, this.outputFilename);
         try {
             return ByteBuffer.wrap(readFile(this.bucketName, this.outputFilename));
         } catch (MinioException | IOException | NoSuchAlgorithmException | InvalidKeyException e) {
             LOG.error("Error reading output file");
             throw new StorageReadException(e.getMessage());
         }
     }
-}
+
+    private void saveData(ByteBuffer byteBuffer, String bucketName, String filename) throws StorageWriteException, StorageReadException {
+
+        try {
+            if (!bucketExists(bucketName)) {
+                minioClient.makeBucket(MakeBucketArgs.builder().bucket(bucketName).build());
+            }
+            final String data = new String(byteBuffer.array(), StandardCharsets.UTF_8);
+            final InputStream inputStream = new ByteArrayInputStream(data.getBytes());
+
+            minioClient.putObject(
+                    PutObjectArgs.builder()
+                            .bucket(bucketName)
+                            .object(filename)
+                            .stream(inputStream, data.length(), -1).build());
+        } catch (MinioException | NoSuchAlgorithmException | IOException | InvalidKeyException e) {
+            throw new StorageWriteException(e.getMessage());
+        } catch (StorageReadException e) {
+            throw new StorageReadException(e.getMessage());
+        }
+    }
+
+    private void appendData(ByteBuffer byteBuffer, String bucketName, String filename) throws StorageWriteException {
+        final String tempFilename = "tmp-" + UUID.randomUUID();
+        final List<ComposeSource> sources = List.of(
+                ComposeSource.builder().bucket(bucketName).object(filename).build(),
+                ComposeSource.builder().bucket(bucketName).object(tempFilename).build());
+        try {
+            saveData(byteBuffer, bucketName, tempFilename);
+            minioClient.composeObject(ComposeObjectArgs.builder().bucket(bucketName).object(filename)
+                    .sources(sources).build());
+            // Delete temporary file
+            minioClient.removeObject(RemoveObjectArgs.builder().bucket(bucketName).object(tempFilename).build());
```

Comment:
In theory all of this should be thread safe to avoid race conditions. I would suggest for now to add a `synchronized` keyword to the whole method, knowing that this is going to have impact on performance and that it works only if there is a single replica running